### PR TITLE
IDV Award type API contract updates

### PR DESCRIPTION
**Pull request by kojodoesdesign**
**High level description:**
Updates API Contract fields in preparation for IDV data types

**JIRA Ticket:**
[DEV-1806](https://federal-spending-transparency.atlassian.net/browse/DEV-1806)

The following are ALL required for the PR to be merged:
- [x]  Contract approved by backend (@tony-sappe / @justinle )

### CODE REVIEW

File: api_contracts/contracts/search/AdvancedSearch.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ This endpoint returns a list of aggregated award amounts grouped by time period
 + recipient_locations (optional, array[LocationObject])
 + recipient_type_names: `category_business` (optional, array[string])
     See options at https://github.com/fedspendingtransparency/usaspending-api/wiki/Recipient-Business-Types
-+ award_type_codes: A, B, C, D (optional, array[string])
++ award_type_codes: 01, 02, 03, 04 (optional, array[string])
```
Comment (tony-sappe) [inline api_contracts/contracts/search/AdvancedSearch.md]: Full list of award type codes with descriptions:
```
    '02': 'Block Grant',
    '03': 'Formula Grant',
    '04': 'Project Grant',
    '05': 'Cooperative Agreement',
    '06': 'Direct Payment for Specified Use',
    '07': 'Direct Loan',
    '08': 'Guaranteed/Insured Loan',
    '09': 'Insurance',
    '10': 'Direct Payment with Unrestricted Use',
    '11': 'Other Financial Assistance',
    'A': 'BPA Call',
    'B': 'Purchase Order',
    'C': 'Delivery Order',
    'D': 'Definitive Contract',
    'E': 'Unknown Type',
    'F': 'Cooperative Agreement',
    'G': 'Grant for Research',
    'S': 'Funded Space Act Agreement',
    'T': 'Training Grant'
```

The current "groups"
```
contract_type_mapping = {
    'A': 'BPA Call',
    'B': 'Purchase Order',
    'C': 'Delivery Order',
    'D': 'Definitive Contract'
}
grant_type_mapping = {
    '02': 'Block Grant',
    '03': 'Formula Grant',
    '04': 'Project Grant',
    '05': 'Cooperative Agreement'
}
direct_payment_type_mapping = {
    '06': 'Direct Payment for Specified Use',
    '10': 'Direct Payment with Unrestricted Use'
}
loan_type_mapping = {
    '07': 'Direct Loan',
    '08': 'Guaranteed/Insured Loan'
}
other_type_mapping = {
    '09': 'Insurance',
    '11': 'Other Financial Assistance'
}
```

File: api_contracts/contracts/recipient/RecipientProfile.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ This endpoint returns a list of recipients, their level, DUNS, and amount.
                 + loans
                 + direct_payments
                 + other_financial_assistance
+                + idv
```
Comment (tony-sappe) [inline api_contracts/contracts/recipient/RecipientProfile.md]: This is out of scope for this sprint

File: api_contracts/contracts/state/StateProfile.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ This endpoint returns the award amounts and totals, based on award type, of a sp
 
 ## StateBreakdown (object)
 + type: contracts (required, string)
-    Award types include 'contracts', 'grants', 'direct_payments', 'loans', 'other_financial_assistance'.
+    Award types include 'contracts', 'grants', 'direct_payments', 'loans', 'other_financial_assistance', 'idv'.
```
Comment (tony-sappe) [inline api_contracts/contracts/state/StateProfile.md]: This should be out-of-scope too

File: api_contracts/contracts/search/AdvancedSearch.md [hunk 4]
```diff
@@ -108,7 +132,9 @@ This endpoint returns a list of aggregated award amounts grouped by time period
 + recipient_locations (optional, array[LocationObject])
 + recipient_type_names: `category_business` (optional, array[string])
     See options at https://github.com/fedspendingtransparency/usaspending-api/wiki/Recipient-Business-Types
-+ award_type_codes: A, B, C, D (optional, array[string])
++ award_type_codes: 02, 03, 04, 05, 06, 07, 08, 09, 10, 11, A, B, C, D, E, F, G, S, T, IDV_A, IDV_B, IDV_C, IDV_D, IDV_E (optional, array[string])
```
Comment (tony-sappe) [inline api_contracts/contracts/search/AdvancedSearch.md]: Since these are strings, do they need backticks on them?
```
`02`,`03`,`04`,`05`,`06`,`07`,`08`,`09`,`10`,`11`,`A`,`B`,`C`,`D`,`E`,`F`,`G`,`S`,`T`,`IDV_A`,`IDV_B`,`IDV_C`,`IDV_D`,`IDV_E`,
```

Comment (kojodoesdesign) [reply api_contracts/contracts/search/AdvancedSearch.md]: yep, just added them!

File: api_contracts/contracts/search/AdvancedSearch.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FORMAT: 1A
+`FORMAT: 1A
```
Comment (tony-sappe) [inline api_contracts/contracts/search/AdvancedSearch.md]: typo?

File: api_contracts/contracts/search/AdvancedSearch.md [hunk 6]
```diff
@@ -7,7 +7,51 @@ These endpoints support the advanced search page and allow for complex filtering
 
 # Group Visualizations
 
-These endpoints return data that is grouped in preset units to support the various data visualizations on USAspending.gov's Advanced Search page.
+These endpoints return data that is grouped in preset units to support the various data visualizations on USAspending.gov`s Advanced Search page.
```
Comment (tony-sappe) [inline api_contracts/contracts/search/AdvancedSearch.md]: >...data that is grouped...
Change to:
data that are grouped

File: api_contracts/contracts/search/AdvancedSearch.md [hunk 7]
```diff
@@ -7,7 +7,51 @@ These endpoints support the advanced search page and allow for complex filtering
...
+
+## Spending by Award [/api/v2/search/spending_by_award/]
+
+These endpoints return data that is grouped in preset units to support the various data visualizations on USAspending.gov`s Advanced Search page.
```
Comment (justinle) [inline api_contracts/contracts/search/AdvancedSearch.md]: Is the usage of backtick `` ` `` rather than of apostrophe `'` intended?
